### ISSUE TITLE
showMatrix with y_array as tree doesn't work well enough

### ISSUE THREAD

**Issue by jamesmkrieger** [prody/ProDy#916]
I get the following error:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-1-e92eb9b2363c> in <module>()
     31
     32 plt.figure()
---> 33 showMatrix(expt_rmsd_matrix, y_array=expt_rmsd_tree)
     34 labels = [label[:4] for label in expt_ens.getLabels()]
     35 plt.xticks(range(expt_ens.numConfs()), labels)

/home/jkrieger/Documents/code/ProDy/prody/utilities/catchall.pyc in showMatrix(matrix, x_array, y_array, **kwargs)
    407
    408             y = y_array
--> 409             xp, yp = interpY(y)
    410             points = np.array([yp, xp]).T.reshape(-1, 1, 2)
    411             segments = np.concatenate([points[:-1], points[1:]], axis=1)

/home/jkrieger/Documents/code/ProDy/prody/utilities/misctools.pyc in interpY(Y)
    451
    452 def interpY(Y):
--> 453     Y = asarray(Y, dtype=float)
    454     n = len(Y)
    455     X = arange(n)

/home/jkrieger/anaconda2/lib/python2.7/site-packages/numpy/core/numeric.pyc in asarray(a, dtype, order)
    499
    500     """
--> 501     return array(a, dtype, copy=False, order=order)
    502
    503

TypeError: float() argument must be a string or a number
```

I think this is because the check for checking whether to check for trees is wrong. On line 328, the `np.isscalar` check fails because a tree isn't a scalar. We might want to use `hasattr` instead.

We also don't really need to import `Phylo` as we have a function `showTree` that's based on `Phylo.draw`.

**Comment by jamesmkrieger** [comment]
To clarify, we end up on line 409 because tree_mode is False.

**Comment by jamesmkrieger** [comment]
We could also use `np.size` instead of `np.isscalar` as a tree has size 1.

**Comment by SHZ66** [comment]
A type checker might be more suitable in this case.

**Comment by jamesmkrieger** [comment]
Yes, maybe we should have our own type checker for trees that doesn't depend on Biopython.

**Comment by jamesmkrieger** [comment]
Using `np.size` does work although now I have another error:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-0ba29fd14ef2> in <module>()
     31
     32 plt.figure()
---> 33 showMatrix(expt_rmsd_matrix, y_array=expt_rmsd_tree)
     34 labels = [label[:4] for label in expt_ens.getLabels()]
     35 plt.xticks(range(expt_ens.numConfs()), labels)

/home/jkrieger/Documents/code/ProDy/prody/utilities/catchall.py in showMatrix(matrix, x_array, y_array, **kwargs)
    402
    403         if tree_mode:
--> 404             Phylo.draw(y_array, do_show=False, axes=ax2, **kwargs)
    405         else:
    406             ax2.set_xticklabels([])

/home/jkrieger/anaconda2/lib/python2.7/site-packages/Bio/Phylo/_utils.pyc in draw(tree, label_func, do_show, show_confidence, axes, branch_labels, label_colors, *args, **kwargs)
    530             getattr(plt, str(key))(**dict(value))
    531         elif not (isinstance(value[0], tuple)):
--> 532             getattr(plt, str(key))(*value)
    533         elif (isinstance(value[0], tuple)):
    534             getattr(plt, str(key))(*value[0], **dict(value[1]))

AttributeError: 'module' object has no attribute 'origin'
```

**Comment by SHZ66** [comment]
I think we need a proper checker because the behavior of np.size on non numpy array objects may change. For this error, simply remove kwargs from line 404 will work.

**Comment by SHZ66** [comment]
It looks like this tree plotting feature is implemented in a rush and not well sorted out. In fact, I don't even remember that I have implemented such feature in showMatrix. There should be a proper checker for kwargs before passing it to Phylo.draw because kwargs is also used for plotting the matrix. We should review this tree plotting feature sometime to make sure we are properly implementing it.

**Comment by jamesmkrieger** [comment]
ok. I'm just putting in fixes for now so that I can use it.

**Comment by jamesmkrieger** [comment]
There are definitely things that can be improved but we have something that works now at least.

**Comment by jamesmkrieger** [comment]
We may be able to use something from #977 for this

**Comment by jamesmkrieger** [comment]
This should be fixed by #995
